cartography/rules/data/frameworks/mitre_attack/requirements/t1098_account_manipulation/__init__.py

```diff
@@ -0,0 +1,317 @@
+from cartography.rules.spec.model import Fact
+from cartography.rules.spec.model import Module
+from cartography.rules.spec.model import Requirement
+
+# AWS
+_aws_account_manipulation_permissions = Fact(
+    id="aws_account_manipulation_permissions",
+    name="IAM Principals with Account Creation and Modification Permissions",
+    description=(
+        "AWS IAM users and roles with permissions to create, modify, or delete IAM "
+        "accounts and their associated policies."
+    ),
+    cypher_query="""
+        MATCH (a:AWSAccount)-[:RESOURCE]->(principal:AWSPrincipal)
+        MATCH (principal)-[:POLICY]->(:AWSPolicy)-[:STATEMENT]->(stmt:AWSPolicyStatement)
+        WHERE NOT principal.name STARTS WITH 'AWSServiceRole'
+        AND NOT principal.name CONTAINS 'QuickSetup'
+        AND principal.name <> 'OrganizationAccountAccessRole'
+        AND stmt.effect = 'Allow'
+        WITH a, principal, stmt,
+            // Return labels that are not the general "AWSPrincipal" label
+            [label IN labels(principal) WHERE label <> 'AWSPrincipal'][0] AS principal_type,
+            // Define the list of IAM actions to match on
+            [p IN ['iam:Create','iam:Attach','iam:Put','iam:Update','iam:Add'] |
+                p] AS patterns
+        WITH a, principal, principal_type, stmt,
+            // Filter on the desired IAM actions
+            [action IN stmt.action
+                WHERE ANY(prefix IN patterns WHERE action STARTS WITH prefix)
+                OR action = 'iam:*'
+                OR action = '*'
+            ] AS matched_actions
+        // Return only statement actions that we matched on
```

Comment:
In general, the visual queries are much simpler because they don't need to do any text processing. Here, we make sure to return only the actions of the statement that we matched on instead of all the actions of the statement. The visual query doesn't need that logic.